packages/api/src/controllers/taskController.js

```diff
@@ -702,6 +702,14 @@ const taskController = {
         )
         .whereIn('task_id', taskIds);
       const filteredTasks = graphTasks.map(removeNullTypes);
+
+      /* Clean before returning to frontend */
+      filteredTasks.forEach((task) => {
+        if (!task.soil_amendment_task) {
```

Comment:
It is not guaranteed that a soil amendment task will have `soil_amendment_task` property (see the selectId error) . I hope this is the case moving forward but the most reliable way is to check based on type_id.

Reply:
Could you fix it? 😇 

I'm on the frontend for a while with bugs

Reply:
@kathyavini branch or commit?